src/cluster/src/install.rs

```diff
@@ -844,11 +863,55 @@ impl ClusterInstaller {
         Ok(None)
     }
 
+    /// Wait until the Fluvio SC public service appears in Kubernetes
+    async fn wait_for_sc_port_check(&self, addr: &str) -> Result<Option<String>, ClusterError> {
+        for i in 0..30u16 {
+            info!(attempt = i, "waiting for sc port check");
+            let sock_addr_string = format!("{}:9003", addr);
+            let sock_addr = self.wait_for_sc_dns(&sock_addr_string).await?;
+            if let Ok(_) = TcpStream::connect(&*sock_addr).await {
+                return Ok(Some(sock_addr_string));
+            }
+            let sleep_ms = 1000 * 2u64.pow(i as u32);
+            info!(
+                attempt = i,
+                "sc port 9003 closed, sleeping for {} ms", sleep_ms
+            );
+            sleep(Duration::from_millis(sleep_ms)).await
+        }
+
+        Ok(None)
+    }
+
+    /// Wait until the Fluvio SC public service appears in Kubernetes
+    async fn wait_for_sc_dns(
+        &self,
+        sock_addr_string: &str,
+    ) -> Result<Vec<SocketAddr>, ClusterError> {
+        for i in 0..30u16 {
+            info!(attempt = i, "waiting for sc dns resolution");
+            match resolve(sock_addr_string).await {
+                Ok(sock_addr) => return Ok(sock_addr),
+                Err(err) => {
+                    let sleep_ms = 1000 * 2u64.pow(i as u32);
+                    info!(
+                        attempt = i,
+                        "dns resoultion failed {}, sleeping for {} ms", err, sleep_ms
+                    );
+                    sleep(Duration::from_millis(sleep_ms)).await
+                }
+            }
+        }
+
+        Err(ClusterError::Other("unable to resolve DNS".to_owned()))
```

Comment:
Should we have a special error type for this?

Reply:
Yes, maybe `SC_NOT_REACHABLE`

Reply:
Will add error type